snapcraft.yaml

```diff
@@ -90,3 +61,122 @@ plugs:
     interface: content
     target: $SNAP/data-dir/sounds
     default-provider: gtk-common-themes
+  brave-config:
+    interface: personal-files
+    read: [$HOME/.config/brave]
+parts:
+  brave:
+    plugin: dump
+    source: https://github.com/brave/brave-browser/releases/download/v$SNAPCRAFT_PROJECT_VERSION/brave-browser_$SNAPCRAFT_PROJECT_VERSION_amd64.deb
+    after: [desktop-gtk3, gtk-key-themes]
+    stage-packages:
+      - libgbm1
+      - libgl1-mesa-glx
+      - libgtk-3-0
+      - libnss3
+      - libsecret-1-0
+      - libxss1
+      - pulseaudio
+      - shared-mime-info
+      - ubuntu-settings
+      - fonts-liberation
+    override-pull: |
+      set -eux
+      snapcraftctl pull
+      rm -rf etc/cron.daily/ usr/bin/brave-browser usr/bin/brave-browser-stable
+      sed -i 's|Icon=brave-browser|Icon=/opt/brave.com/brave/product_logo_128\.png|g' usr/share/applications/brave-browser.desktop
+    prime:
+      - -etc/gss
+      - -etc/init.d
+      - -etc/sensors.d
+      - -etc/ucf.conf
+      - -etc/xdg
+      - -usr/include
+      - -usr/lib/tmpfiles.d
+      - -usr/share/apport
+      - -usr/share/bash-completion
+      - -usr/share/bug
+      - -usr/share/doc
+      - -usr/share/doc-base
+      - -usr/share/gettext
+      - -usr/share/gnome-control-center
+      - -usr/share/GConf
+      - -usr/share/icons
+      - -usr/share/lintian
+      - -usr/share/man
+      - -usr/share/pam-configs
+      - -usr/share/pkgconfig
+      - -usr/share/polkit-1
+      - -usr/share/session-migration
+      - -usr/share/sounds
+      - -usr/share/ubuntu
+      - -usr/share/ubuntu-wayland
+      - -usr/share/upstart
+      - -usr/share/zsh
+      - -var
+      - -opt/brave.com/brave/chrome-sandbox
+  desktop-gtk3:
+    source: https://github.com/ubuntu/snapcraft-desktop-helpers.git
+    source-subdir: gtk
+    plugin: make
+    make-parameters: ["FLAVOR=gtk3"]
+    build-packages:
+      - build-essential
+      - libgtk-3-dev
+    stage-packages:
+      - libxkbcommon0
+      - ttf-ubuntu-font-family
+      - shared-mime-info
+      - libgtk-3-0
+      - libgdk-pixbuf2.0-0
+      - libglib2.0-bin
+      - libgtk-3-bin
+      - unity-gtk3-module
+      - locales-all
+      - xdg-user-dirs
+      - ibus-gtk3
+      - libibus-1.0-5
+      - fcitx-frontend-gtk3
+    prime:
+      - -etc/gss
+      - -etc/presage.xml
+      - -etc/ucf.conf
+      - -etc/X11
+      - -usr/lib/glib-networking
+      - -usr/lib/systemd
+      - -usr/share/apport
+      - -usr/share/bash-completion
+      - -usr/share/doc
+      - -usr/share/doc-base
+      - -usr/share/gettext
+      - -usr/share/GConf
+      - -usr/share/icons
+      - -usr/share/lintian
+      - -usr/share/man
+      - -usr/share/pkgconfig
+      - -usr/share/presage
+      - -usr/share/upstart
+      - -var
+  shared-mime-info:
+    after: [brave]
+    plugin: nil
+    override-pull: ""
+    override-prime: |
+      set -eux
```

Comment:
i never felt these set commands were really needed. Just makes things super verbose for debugging, but might not be great for production. Maybe just leave eu and leave off debugging (x)

Reply:
Do we rant to return       chmod 0755 ./opt/brave.com/brave/chrome-sandbox
. No reason to have a setuid sandbox for a snap

Reply:
I think?

Reply:
@posix4e Thanks for the suggestions! Good catch on the `chrome-sandbox` folder.

Updated in 934e3eb.